plugins/org.jkiss.dbeaver.ext.gaussdb/plugin.xml

```diff
@@ -0,0 +1,541 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?eclipse version="3.2"?>
+<plugin>
+    <extension point="org.jkiss.dbeaver.postgresql.serverType">
+        <serverType id="gaussdb"
+                    name="GaussDB"
+                    class="org.jkiss.dbeaver.ext.gaussdb.model.PostgreServerGaussDB"
+                    logo="icons/gaussdb_logo.png"
+                    customURL="true" turnOffPreparedStatements="true"/>
+    </extension>
+    
+    <extension point="org.jkiss.dbeaver.sqlDialect">
+        <dialect id="gaussdb" parent="basic" class="org.jkiss.dbeaver.ext.gaussdb.model.GaussDBDialect" label="GaussDB" description="GaussDB dialect." icon="icons/gaussdb_icon.png">
+            <property name="insertMethods" value="pgInsertIgnore,pgReplaceIgnore"/>
+            <keywords value=""/>
+            <execKeywords value=""/>
+            <ddlKeywords value=""/>
+            <dmlKeywords value=""/>
+            <functions value=""/>
+            <types value=""/>
+
+            <property name="" value=""/>
+        </dialect>
+    </extension>
+
+    <extension point="org.jkiss.dbeaver.dataSourceProvider">
+        <datasource
+                class="org.jkiss.dbeaver.ext.gaussdb.GaussDBDataSourceProvider"
+                description="%datasource.gaussdb.description"
+                id="gaussdb"
+                label="GaussDB"
+                icon="icons/gaussdb_icon.png"
+                dialect="gaussdb"
+                inheritClients="false">
+                
+            <drivers managable="true">
+                <driver
+                        id="gaussdb-jdbc"
+                        label="GaussDB"
+                        icon="icons/gaussdb_icon.png"
+                        iconBig="icons/gaussdb_icon_big.png"
+                        class="org.postgresql.Driver"
+                        sampleURL="jdbc:postgresql://{host}[:{port}]/[{database}]"
+                        defaultPort="8000"
+                        
+                        description="%driver.gaussdb.description"
+                        categories="sql,analytic">
+
+                    <file type="jar" path="https://repo.huaweicloud.com/repository/maven/huaweicloudsdk/gsjdbc4/gsjdbc4/1.0/gsjdbc4-1.0.jar"
+                          bundle="!drivers.gaussdb"/>
+                    <parameter name="serverType" value="gaussdb"/>
+                    <property name="loginTimeout" value="20"/>
+                    <property name="connectTimeout" value="20"/>
+                    <property name="prepareThreshold" value="0"/>
+                </driver>
+            </drivers>
+            
+
+          <tree path="gaussdb" label="GaussDB data source">
```

Comment:
Please use treeInjection instead of copy pasting the code. You can take a look at greenpulm plugin

Reply:
We have modified the postgresql directory tree as follows:
1. Schemas are divided into user schemas and system schemas.
![schemas](https://github.com/dbeaver/dbeaver/assets/172629283/51f5827d-e3ad-4e37-aa3d-3a99442334d3)

3. The function is divided into function and procedure. In addition, the Oracle database in compatibility mode supports the package.
![image](https://github.com/dbeaver/dbeaver/assets/172629283/fbbfe7eb-03ca-4f9f-aa07-ef236eddaa0e)

If the treeInjection mode is used to inherit the plugin.xml file of the PostgreSQL database, you still need to perform most of the repeated configurations in the plugin.xml file.

Reply:
Please review this review opinion. Thank you @E1izabeth 
Note: We will further make customized modifications in the future.

Reply:
1. We can not accept this copy-paste, because such code is badly supportable.
This way you can remove something from the tree
```
<treeInjection
      path="postgresql/database/users" remove="true">
</treeInjection>
```
another example of changing class responsible for objects in folder
```
<treeInjection
                    path="generic/catalog/schema/table"
                    changeFolderType="path.to.class.MyDbTable"/>
```
  to separate procedures and functions
```
              <treeInjection path="mydb/schema/procedure" changeFolderType="path.to.MyDbProcedure"/>
            <treeInjection path="mydb/schema/function" changeFolderType="path.to.MyDbFunction"/>
```
2. We already have the concept of system and user schemas. System schemas come if "Show system objects" is turned on. All you need is to override `isSystem` method on class described schema object.
![image](https://github.com/dbeaver/dbeaver/assets/28875055/146a032f-a4cd-411c-a60b-6726652cdf01)
Current solution broke this DBeaver concept. We can't accept it.

Reply:
We have made all the modifications and self verified several functional points according to your suggestions, and will submit them later today.

Reply:
We have modified and tested according to your feedback. Please take up your valuable time for further inspection? Thank you~ @E1izabeth

Reply:
Thank you